src/sql/src/plan/statement/ddl.rs

```diff
@@ -911,7 +916,58 @@ pub fn plan_create_source(
             if key_envelope == KeyEnvelope::None {
                 key_envelope = get_unnamed_key_envelope(key_encoding)?;
             }
-            UnplannedSourceEnvelope::Upsert(UpsertStyle::Default(key_envelope))
+
+            let order_by = if !order_by.is_empty() {
+                scx.require_unsafe_mode("ENVELOPE UPSERT (ORDER BY ...)")?;
+
+                let mut offset_included = false;
+
+                let metadata_indices = order_by
+                    .iter()
+                    .map(|order_col| {
+                        let metadata_type = metadata_columns
+                            .iter()
+                            .position(|(col_name, _)| *col_name == order_col.as_str())
+                            .map(|pos| {
+                                let included_col = metadata_column_types
+                                    .get(pos)
+                                    .expect("metadata index calculation error");
+                                (pos, included_col)
+                            });
+
+                        match metadata_type {
+                            Some((pos, &IncludedColumnSource::Timestamp)) => Ok(pos),
+                            Some((pos, &IncludedColumnSource::Offset)) => {
+                                offset_included = true;
+                                Ok(pos)
+                            }
+                            Some(_) => sql_bail!(
+                                "Only `TIMESTAMP` and `OFFSET` (from `INCLUDE TIMESTAMP` and `INCLUDE OFFSET`) are supported as `ORDER BY` columns"
+                            ),
+                            None => sql_bail!("Could not find {} in included metadata", order_col),
```

Comment:
This message does not seem to be informative enough - For example, the following two produce an identical message:

```
materialize=>  CREATE SOURCE texttext
  FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-textbytes-${testdrive.seed}')
  KEY FORMAT TEXT
  VALUE FORMAT TEXT
  INCLUDE OFFSET as o, TIMESTAMP as ts
  ENVELOPE UPSERT ( ORDER BY ( o ASC, no_such_key_exists_ever ASC ));
ERROR:  Could not find some_key in included metadata
materialize=>  CREATE SOURCE texttext
  FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-textbytes-${testdrive.seed}')
  KEY FORMAT TEXT
  VALUE FORMAT TEXT
  INCLUDE OFFSET as o, TIMESTAMP as ts
  ENVELOPE UPSERT ( ORDER BY ( o ASC, key ASC ));
ERROR:  Could not find key in included metadata
```

I think the `"Only `TIMESTAMP` and `OFFSET` (from `INCLUDE TIMESTAMP` and `INCLUDE OFFSET`) are supported as `ORDER BY` columns"` is more informative and should also be returned in those two cases.

Reply:
Ack, will update 👍